### ISSUE TITLE
Create method like doOnNext(func) where func is returning observable

### ISSUE THREAD

**Issue by ellenaua** [Reactive-Extensions/RxJS#1317]
In many places of my project 
I need to do some async actions for each message. 
but pass original message forward

Currently I have written this:

Rx.Observable.from([...])
.flatMap(data => doSomething$(data), (a, b) => a)
.flatMap(data => doSomethingElse$(data), (a, b) => a)
.flatMap(data => doSomethingOther$(data), (a, b) => a)

but this "(a, b) => a" is annoying

Is there some method which makes asyncronous action but passes forward original message?

**Comment by ellenaua** [comment]
Basically, it's very easy to create such method - maybe it already exists?

Rx.Observable.prototype.flatMapReturnOriginal = function (func) {
  return this.flatMap(func, (a, b) => a);
}

**Comment by paulpdaniels** [comment]
I don't think it makes sense to add a new operator for this, since I don't think it is a super common use case (I could be wrong though).

I would suggest using `let` instead

```javascript

function flatMapForward(func) {
  return source => source.flatMap(func, a => a)
}

Rx.Observable.from([])
  .let(forward(doSomething))
  .let(forward(doSomethingElse))
  .let(forward(doSomethingOther))
  
  .subuscribe

```
